File: app/src/main/java/me/imzack/app/end/view/contract/PlanSearchViewContract.java
package me.imzack.app.end.view.contract;

import me.imzack.app.end.view.adapter.PlanSearchListAdapter;

public interface PlanSearchViewContract extends BaseViewContract {

    void showInitialView(int planCount, PlanSearchListAdapter planSearchListAdapter);

    void onSearchChanged(boolean isNoSearchInput, boolean isPlanSearchEmpty);

    void onPlanItemClicked(int planListPos);
}

File: app/src/main/kotlin/me/imzack/app/end/view/contract/PlanSearchViewContract.kt
package me.imzack.app.end.view.contract

import me.imzack.app.end.view.adapter.PlanSearchListAdapter

interface PlanSearchViewContract : BaseViewContract {

    fun showInitialView(planCount: Int, planSearchListAdapter: PlanSearchListAdapter)

    fun onSearchChanged(isNoSearchInput: Boolean, isPlanSearchEmpty: Boolean)

    fun onPlanItemClicked(planListPos: Int)
}
